### PR TITLE
Patrick/adding cart button

### PR DESCRIPTION
**Pull request by patrick-fs**
I've added a "My Cart" button that displays the size of the current cart. I'm using the Redux `useSelector` hook to update state.

@van-fs I'm going to ask Tim to review this one to ensure the Redux-y things look good and then assign to you once it LGTT.

### CODE REVIEW

File: src/components/CartButton.tsx [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+import { Button } from 'react-bootstrap';
+import { useSelector } from 'react-redux';
+import { Link } from 'react-router-dom';
+import { AppState } from '../store';
+import { Product } from '../types/product';
+
+const CartButton = () => {
+ 
+  const products = useSelector<AppState, Product[]>(state => state.cartState.products);
```
Comment (timjchin) [inline src/components/CartButton.tsx]: This is definitely valid, but one thing that can be a pain as it grows is having to supply the second generic (shape of the return from the selector). Instead, we often rely on inference:

```ts
  const products = useSelector((state: AppState) => state.cartState.products);
```

**Review by timjchin — APPROVED** [review]
This LGTM! My apologies for the delay.

File: src/components/CartButton.tsx (hunk 1 above)
Comment (patrick-fs) [reply src/components/CartButton.tsx]: Thanks Tim, I'll fix this up.

**Review by van-fs — APPROVED** [review]
LGTM

**Pull request merged**